src/gui/pages_maped/KM_GUIMapEdUnit.pas

```diff
@@ -149,8 +149,16 @@ procedure TKMMapEdUnit.Show(aGroup: TKMUnitGroup);
 
 procedure TKMMapEdUnit.Unit_ArmyChange1(Sender: TObject);
 begin
-  if Sender = Button_Army_ForUp then fGroup.UnitsPerRow := fGroup.UnitsPerRow - 1;
-  if Sender = Button_Army_ForDown then fGroup.UnitsPerRow := fGroup.UnitsPerRow + 1;
+  Unit_ArmyChangeShift(Sender, []);
+end;
+
+
+procedure TKMMapEdUnit.Unit_ArmyChangeShift(Sender: TObject; Shift: TShiftState);
+begin
+  if Sender = Button_Army_ForUp then
+    fGroup.UnitsPerRow := EnsureRange(fGroup.UnitsPerRow - GetMultiplicator(Shift), 1, 200);
+  if Sender = Button_Army_ForDown then
+    fGroup.UnitsPerRow := EnsureRange(fGroup.UnitsPerRow + GetMultiplicator(Shift), 1, 200);
```

Comment:
`EnsureRange` looks superfluous here. `UnitsPerRow` already ensures the value is within 1..Count in its setter. Why do we need it here?